example/script/src/main.rs

```diff
@@ -1,172 +1,20 @@
-pub const ISPRIME_ELF: &[u8] = include_bytes!("../../elfs/isprime-riscv32im-succinct-zkvm-elf");
-pub const SHA2_ELF: &[u8] = include_bytes!("../../elfs/sha2-riscv32im-succinct-zkvm-elf");
-pub const TENDERMINT_ELF: &[u8] =
-    include_bytes!("../../elfs/tendermint-riscv32im-succinct-zkvm-elf");
-
-pub const GROTH16_VK_BYTES: &[u8] = include_bytes!("../../vk/groth16_v2.0.0.bin");
-
-#[derive(clap::Parser)]
-#[command(name = "zkVM Proof Generator")]
-struct Cli {
-    #[arg(
-        long,
-        value_name = "ELF",
-        default_value = "fibonacci",
-        help = "Specifies the ELF file to use (e.g., fibonacci, is-prime)"
-    )]
-    elf: String,
-    #[arg(
-        long,
-        value_name = "prove",
-        default_value = "false",
-        help = "Specifies the ELF file to use (e.g., fibonacci, is-prime)"
-    )]
-    prove: bool,
-}
-
-#[derive(Debug, EnumString, EnumIter, Display)]
-enum Elf {
-    #[strum(serialize = "fibonacci")]
-    Fibonacci,
-    #[strum(serialize = "is-prime")]
-    IsPrime,
-    #[strum(serialize = "sha2")]
-    Sha2,
-    #[strum(serialize = "tendermint")]
-    Tendermint,
-}
-
-impl Elf {
-    fn get_elf(&self) -> &'static [u8] {
-        match self {
-            Elf::Fibonacci => FIBONACCI_ELF,
-            Elf::IsPrime => ISPRIME_ELF,
-            Elf::Sha2 => SHA2_ELF,
-            Elf::Tendermint => TENDERMINT_ELF,
-        }
-    }
-}
+use groth16_solana::{verify_proof_fixture, SP1ProofFixture, GROTH16_VK_BYTES};
+use sp1_sdk::{utils, SP1ProofWithPublicValues};
 
 fn main() {
     // Setup logging for the application.
     utils::setup_logger();
 
-    // Parse command line arguments.
-    let args = Cli::parse();
-    let mut stdin = SP1Stdin::new();
-
-    let elf_enum = Elf::from_str(&args.elf)
-        .expect("Invalid ELF name. Use 'fibonacci', 'is-prime', or other valid ELF names.");
-    let elf = match elf_enum {
-        Elf::Fibonacci => {
-            let n = 20;
-            stdin.write(&n);
-            elf_enum.get_elf()
-        }
-        Elf::IsPrime => {
-            let n = 11u64;
-            stdin.write(&n);
-            elf_enum.get_elf()
-        }
-        Elf::Sha2 | Elf::Tendermint => elf_enum.get_elf(),
-    };
-
     // Where to save / load the proof from.
-    let proof_file = format!("../binaries/{}_proof.bin", args.elf);
-
-    // Only generate a proof if the prove flag is set.
-    if args.prove {
-        // Initialize the prover client
-        let client = ProverClient::new();
-        let (pk, _) = client.setup(elf);
+    let proof_file = "../../proofs/fibonacci_proof.bin";
 
-        // Generate a proof for the specified program.
-        let proof = client
-            .prove(&pk, stdin)
-            .groth16()
-            .run()
-            .expect("Groth16 proof generation failed");
-
-        // Save the generated proof to `proof_file`.
-        proof.save(&proof_file).unwrap();
-    }
```

Comment:
Note: I would keep this in the script. 

Your `README.md` should have the user just run one script and show the stages.

1. Load SP1 Proof. If `--prove` flag is passed, generate a new Groth16 proof from the network that will take 1 minute to generate a proof.
2. Save the borsh-serialized `SP1ProofFixture`.
3. Run a Rust script to create a transaction that's sent to the Solana devnet that loads the borsh-serialized `SP1ProofFixture`.
4. This transaction will hit a Solana program that verifies the borsh-serialized SP1ProofFixture.